### PR TITLE
OD-106 [Feature] Users will see selected filters that they have select in the filter overlay

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-106

## Description
We have added a new function into the utils.js file to handle when we should show users active filters.

## Screenshots/screencasts
https://share.getcloudapp.com/YEuQ2YPk

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko @ivandevupp

### CODE REVIEW

File: js/utils.js [hunk 1]
```diff
@@ -927,6 +927,82 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
...
+        toggle: this.dataset.toggle,
+        field: this.dataset.field,
+        value: this.dataset.value
+      });
+
+      activeFiltersElements.push(activeFilter);
+    });
+
+    $filtersGroup.html(activeFiltersElements);
+    $filtersGroup.on('click', '.hidden-filter-controls-filter.mixitup-control-active', options, onActiveFilterClick);
```
Comment (tonytlwu) [inline js/utils.js:1002]: This is possibly why you were having problem with multiple click events. The filter group is always on the page, which means there's no need to add the event listeners every time the active filters are updated. What I've suggested here is to add the event listeners to the filter options, which avoids having duplicated event listeners on a single element.

```suggestion
    $filtersGroup.find('.hidden-filter-controls-filter.mixitup-control-active').on('click', function() {
      onActiveFilterClick(options);
    });
```

File: js/utils.js [hunk 2]
```diff
@@ -927,6 +927,82 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
...
+
+  /**
+   * This function designed to show users filters that were activated in the filters overlay
+   *
+   * @param {Object} options - and object with keys:
+   *   context { Object } - the layout instance
+   *   filtersInOverlay { Boolean } - boolean variable that tells us if filters shown in overlay
+   *   filterOverlayClass { String } - a CSS class of the layout filter overlay
+   *
+   * @return {void} this function doesn't return anything it adds active filters directly into the DOM of the page
```
Comment (tonytlwu) [inline js/utils.js:972]: You can just leave it out.

```suggestion
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Comment by YaroslavOvdii** [comment]
@tonytlwu added suggestions

File: js/utils.js [hunk 3]
```diff
@@ -985,37 +977,37 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
...
 
       return;
     }
 
-    $selectedFilters.each(function() {
-      var activeFilter = getActiveFilterNode({
-        toggle: this.dataset.toggle,
-        field: this.dataset.field,
-        value: this.dataset.value
+    var activeFiltersElements = $.map($selectedFilters, function(filter) {
```
Comment (tonytlwu) [inline js/utils.js:998]:
```suggestion
    var activeFilterElements = $.map($selectedFilters, function(filter) {
```

File: js/utils.js [hunk 4]
```diff
@@ -927,6 +927,82 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
...
+   *
+   * @return {void} this function doesn't return anything it adds active filters directly into the DOM of the page
+   */
+  function addActiveFilters(options) {
+    if (!options.filtersInOverlay) {
+      return;
+    }
+
+    var $container = options.context.$container;
+    var $selectedFilters = $container.find('[data-filter-group] .mixitup-control-active');
```
[the remark this answers is not in the document]
Comment (tonytlwu) [reply js/utils.js:988]: This was not done

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Comment by YaroslavOvdii** [comment]
@tonytlwu Renamed variables

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: LFD users with filters enabled
Verify: LFD filters are working as expected

Squash and merge: Show active filters when filters are selected in overlay

**Pull request merged**